TicketUtil.py

```diff
@@ -119,6 +118,113 @@ def close_requests_session(self):
         """
         self.s.close()
 
+class BugzillaTicket(Ticket):
+    """
+    A BZ Ticket object. Contains BZ-specific methods for working with tickets.
+    """
+
+    def __init__(self, base_url, project_key, ticket_id=None, auth=None, user=None, password=None):
+        self.ticketing_tool = 'Bugzilla'
+
+        # Right now, hardcode auth as 'kerberos', which is the only supported auth for BZ.
+        self.auth = auth if auth else 'kerberos'
+
+        self.credentials = {"login": user, "password": password}
+        self.token = None
+
+        # BZ URLs
+        self.base_url = base_url
+        self.rest_url = '{0}/rest/bug'.format(self.base_url)
+        self.auth_url = '{0}/rest/login'.format(self.base_url)
+
+        # Call our parent class's init method which creates our requests session.
+        super(BugzillaTicket, self).__init__(project_key, ticket_id)
+
+
+    def create_ticket_parameters(self, options_dict):
+
+        params = {"product": str(self.project_key)}
+
+        # Iterate through our options and add them to the params dict.
+        for key, value in options_dict.items():
```

Comment:
Huh, I'm thinking, why you don't just do `params.update(options_dict)` instead of iterating through each item in `options_dict` and updating the `params` separately?

For learning purposes: iteration through items on lines 149 and 150 does following:
1. builds a list of tuples (a tuple for each item)
2. creates a dict (containing one key/value pair `{str(key): str(value)}`)
3. updates the `params` dict.
4. repeat for next tuple

From memory point of view:
- the step 1 allocates memory for the list and each tuple
- step no. 2 allocates a new dict
- step no. 3 can result into memory reallocation since the dict needs to fit in the added item. And this can happened multiple times for each item parsed.

So in theory you can end up with a lot of (re)allocations there... In real world the python interpret would transcribe this for you and it would actually run something similar to `params.update(options_dict)`.

That being said, you can leave the code as is, and it would work just fine. In case you want to make life easier for others reading the code ( the method than becomes a self-explaining 2 liner) and for the python interpret as well, I suggest you to go for `params.update(options_dict)`.